### ISSUE TITLE
Error validating transaction

### ISSUE THREAD

**Issue by yinjun1991** [wavesplatform/Waves#2822]
**Describe the bug**
I start a new waves fullnode, the current height is `1843402`, obvious is not the latest height, maybe the node is syncing. I found some error message in log. Is it normal when syncing ?
```
Dec 18 03:01:10 ip-172-31-3-158 java[8762]: java.lang.IllegalMonitorStateException: attempt to unlock read lock, not locked by current thread
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.locks.ReentrantReadWriteLock$Sync.unmatchedUnlockException(ReentrantReadWriteLock.java:444)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.locks.ReentrantReadWriteLock$Sync.tryReleaseShared(ReentrantReadWriteLock.java:428)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.locks.AbstractQueuedSynchronizer.releaseShared(AbstractQueuedSynchronizer.java:1341)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.locks.ReentrantReadWriteLock$ReadLock.unlock(ReentrantReadWriteLock.java:881)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.BlockchainUpdaterImpl.accountData(BlockchainUpdaterImpl.scala:51)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.transaction.smart.WavesEnvironment.$anonfun$data$2(WavesEnvironment.scala:56)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.transaction.smart.WavesEnvironment.data(WavesEnvironment.scala:44)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.transaction.smart.WavesEnvironment.data(WavesEnvironment.scala:22)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.lang.v1.evaluator.ctx.impl.EnvironmentFunctions.$anonfun$getData$7(EnvironmentFunctions.scala:34)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.lang.v1.evaluator.ctx.impl.EnvironmentFunctions.getData(EnvironmentFunctions.scala:34)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.lang.v1.evaluator.ctx.impl.waves.Functions$$anon$1.apply(Functions.scala:39)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.lang.v1.evaluator.ctx.NativeFunction.eval(NativeFunction.scala:43)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.lang.v1.evaluator.EvaluatorV1.$anonfun$evalFunctionCall$9(EvaluatorV1.scala:100)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.lang.v1.task.TaskMT.$anonfun$flatMap$2(TaskMT.scala:38)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at cats.Eval$.loop$1(Eval.scala:363)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at cats.Eval$.cats$Eval$$evaluate(Eval.scala:368)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at cats.Eval$FlatMap.value(Eval.scala:307)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.lang.v1.evaluator.EvaluatorV1.evalWithLogging(EvaluatorV1.scala:159)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.lang.v1.evaluator.ContractEvaluator$.apply(ContractEvaluator.scala:90)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.InvokeScriptTransactionDiff$.$anonfun$apply$10(InvokeScriptTransactionDiff.scala:91)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.InvokeScriptTransactionDiff$.$anonfun$apply$8(InvokeScriptTransactionDiff.scala:79)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.InvokeScriptTransactionDiff$.$anonfun$apply$2(InvokeScriptTransactionDiff.scala:78)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.InvokeScriptTransactionDiff$.apply(InvokeScriptTransactionDiff.scala:57)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.TransactionDiffer$.$anonfun$unverified$1(TransactionDiffer.scala:73)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.TransactionDiffer$.unverified(TransactionDiffer.scala:69)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.TransactionDiffer$.$anonfun$verified$9(TransactionDiffer.scala:56)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.TransactionDiffer$.$anonfun$verified$1(TransactionDiffer.scala:43)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.TransactionDiffer$.verified(TransactionDiffer.scala:42)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.TransactionDiffer$.$anonfun$apply$1(TransactionDiffer.scala:32)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.state.diffs.TransactionDiffer$.apply(TransactionDiffer.scala:34)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.utx.UtxPoolImpl.com$wavesplatform$utx$UtxPoolImpl$$addTransaction(UtxPoolImpl.scala:173)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.utx.UtxPoolImpl.$anonfun$putNewTx$13(UtxPoolImpl.scala:154)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.utx.UtxPoolImpl.putNewTx(UtxPoolImpl.scala:154)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.utx.UtxPoolImpl.putIfNew(UtxPoolImpl.scala:59)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.network.UtxPoolSynchronizer$.$anonfun$apply$1(UtxPoolSynchronizer.scala:88)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.network.UtxPoolSynchronizerImpl.$anonfun$validateFuture$1(UtxPoolSynchronizer.scala:63)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at scala.util.Success.$anonfun$map$1(Try.scala:255)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at scala.util.Success.map(Try.scala:213)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at monix.execution.internal.InterceptRunnable.run(InterceptRunnable.scala:27)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at com.wavesplatform.utils.Schedulers$TimedWrapper.run(Schedulers.scala:95)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Dec 18 03:01:10 ip-172-31-3-158 java[8762]:         at java.lang.Thread.run(Thread.java:748)
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Version: 1.1.5 and 1.1.4

**Comment by ksemaev** [comment]
I have absolutely the same behaviour on onr of the 3 nodes. I even tried to restore it's storage from working node - still I get the same error

**Comment by br0x** [comment]
This one is already fixed

**Comment by phearnot** [comment]
Fixed in #2696 and released with 1.1.6. Further improvements were implemented in #2885 and will be released with 1.1.8.
